app/src/main/java/org/mozilla/fenix/compose/button/Button.kt

```diff
@@ -31,55 +29,213 @@ import org.mozilla.fenix.theme.FirefoxTheme
 import org.mozilla.fenix.theme.Theme
 
 /**
- * Primary button.
+ * Base component for buttons.
  *
  * @param text The button text to be displayed.
- * @param icon Optional button icon to be displayed.
+ * @param contentColor [Color] to apply to the button text.
+ * @param backgroundColor The background [Color] of the button.
+ * @param icon Optional [Painter] used to display a [Icon] before the button text.
+ * @param tint Tint [Color] to be applied to the icon.
  * @param onClick Invoked when the user clicks on the button.
  */
 @Composable
-fun PrimaryButton(
+fun Button(
     text: String,
+    contentColor: Color,
+    backgroundColor: Color,
     icon: Painter? = null,
+    tint: Color,
     onClick: () -> Unit,
 ) {
-    Button(
+    androidx.compose.material.Button(
         onClick = onClick,
         modifier = Modifier
-            .clip(RoundedCornerShape(size = 4.dp))
             .fillMaxWidth()
             .height(36.dp),
         elevation = ButtonDefaults.elevation(defaultElevation = 0.dp, pressedElevation = 0.dp),
-        colors = ButtonDefaults.outlinedButtonColors(backgroundColor = FirefoxTheme.colors.actionPrimary),
+        colors = ButtonDefaults.outlinedButtonColors(
+            backgroundColor = backgroundColor,
+            contentColor = contentColor,
+        ),
     ) {
-        if (icon != null) {
+        icon?.let { painter ->
             Icon(
-                painter = icon,
+                painter = painter,
                 contentDescription = null,
-                tint = FirefoxTheme.colors.iconOnColor,
+                tint = tint,
             )
 
             Spacer(modifier = Modifier.width(8.dp))
         }
 
         Text(
             text = text,
-            color = FirefoxTheme.colors.textOnColorPrimary,
             fontSize = 14.sp,
-            fontFamily = FontFamily(Font(R.font.metropolis_semibold)),
```

Comment:
why were `color` and `fontFamily` removed?

Reply:
`color` is replaced by the `contentColor`. Removing the `fontFamily` was a flip decision based on the 0.25.sp letterSpacing feedback since the Figma component is technically using 0.25.sp letterSpacing with Roboto whereas our current xml component is using Metropolis Semibold with 0.sp letterSpacing. 

I mostly based my decision here based on the fact that most of the Home screen (most recent was the "See all" link) has already started to convert to using Roboto (system font), and I am looking to make sure we are moving forward with the issue to remove Metropolis.